sql/src/main/java/io/crate/expression/scalar/postgres/CurrentSettingFunction.java

```diff
@@ -0,0 +1,144 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.expression.scalar.postgres;
+
+import io.crate.data.Input;
+import io.crate.expression.scalar.ScalarFunctionModule;
+import io.crate.expression.symbol.Function;
+import io.crate.expression.symbol.Literal;
+import io.crate.expression.symbol.Symbol;
+import io.crate.metadata.BaseFunctionResolver;
+import io.crate.metadata.FunctionIdent;
+import io.crate.metadata.FunctionImplementation;
+import io.crate.metadata.FunctionInfo;
+import io.crate.metadata.FunctionName;
+import io.crate.metadata.Scalar;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.functions.params.FuncParams;
+import io.crate.metadata.functions.params.Param;
+import io.crate.metadata.pgcatalog.PgCatalogSchemaInfo;
+import io.crate.metadata.settings.session.SessionSetting;
+import io.crate.metadata.settings.session.SessionSettingRegistry;
+import io.crate.types.DataType;
+import org.elasticsearch.common.Nullable;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Locale;
+
+public class CurrentSettingFunction extends Scalar<String, Object> {
+
+    private static final String NAME = "current_setting";
+    private static final FunctionName FQN = new FunctionName(PgCatalogSchemaInfo.NAME, NAME);
+
+    public static void register(ScalarFunctionModule scalarFunctionModule) {
+        scalarFunctionModule.register(FQN,
+                                      new BaseFunctionResolver(
+                                          FuncParams
+                                              .builder(Param.STRING)
+                                              .withVarArgs(Param.BOOLEAN)
+                                              .limitVarArgOccurrences(1)
+                                              .build()) {
+                @Override
+                public FunctionImplementation getForTypes(List<DataType> datatypes) {
+                    return new CurrentSettingFunction(
+                        new FunctionInfo(new FunctionIdent(FQN, datatypes), datatypes.get(0))
+                    );
+                }
+            });
+    }
+
+    private final FunctionInfo info;
+
+    CurrentSettingFunction(FunctionInfo info) {
+        this.info = info;
+    }
+
+    @Override
+    public FunctionInfo info() {
+        return info;
+    }
+
+    @Override
+    public Symbol normalizeSymbol(Function function, @Nullable TransactionContext txnCtx) {
```

Comment:
Isn't the default implementation good enough?

Reply:
true, it is - Removed.

I was trying to fail early on the edge cases (`setting = null` or `missingOk = null`) 
but I guess it's not worth it.